src/pf/optimize.lua

```diff
@@ -88,13 +88,12 @@ end
 
 cfkey = memoize(function (expr)
    if type(expr) == 'table' then
-      local parts = {'(', cfkey(expr[1])}
-      for i=2,#expr do
-         parts[i*2-1] = ' '
-         parts[i*2] = cfkey(expr[i])
+      local parts = {'('}
+      for i=1,#expr do
+         parts[i] = cfkey(expr[i])
```

Comment:
Should this be `parts[i+1] = cfkey(expr[i])`? Otherwise the opening parenthesis is overwritten.